pkg/sql/create_type.go

```diff
@@ -186,6 +186,7 @@ func (p *planner) createArrayType(
 		ParentSchemaID: keys.PublicSchemaID,
 		Kind:           sqlbase.TypeDescriptor_ALIAS,
 		Alias:          types.MakeArray(elemTyp),
+		Privileges:     makeDefaultPrivilegeDescriptorForType(params.p.User(), keys.PublicSchemaID),
```

Comment:
I would instead pass a privilege descriptor in for the array type, because it should follow exactly the one used for the main type.